### ISSUE TITLE
[general issue] NaN results produced in `bpp` table (when evolve to HeHG during stable MT)

### ISSUE THREAD

**Issue by TomWagg** [aCOSMIC/aCOSMIC#608]
When running `COSMIC` on large populations of binaries I frequently run into edge cases that result in NaN values in the `bpp` output. I've been chatting with @katiebreivik about this and realised I should probably organise a little. So here's an issue about this and I'll keep a list of commonalities that seem to result in NaNs up to date and post `initC` and `bpp` tables in the comments below.

**NaN clues noticed so far:**

1. This appears to consistently happen when a star evolves on HeHG (kstar=8) _during_ mass transfer
2. Often the `RRLO_1` column is the first one that seems to get a NaN
3. I've only ever seen this happen with binaries (I ran simulations with binfrac=0.5 and kept all of the singles)

edit: GitHub won't let me upload hdf5 files so I'm going to attach separate CSVs instead!

Here's a quick confirmation that it is happening whenever you have a star evolving to HeHG during mass transfer:
```
nan_bpp = pd.read_csv("nans_bpp.csv", index_col=0)
nan_bin_nums = nan_bpp["bin_num"].unique()
hehg_during_mt = [False for _ in range(len(nan_bin_nums))]
for i, bin_num in enumerate(nan_bin_nums):
    single_bpp = nan_bpp.loc[bin_num][["kstar_1", "kstar_2", "evol_type", "RRLO_1", "RRLO_2"]]
    mt = False
    for j in range(len(single_bpp)):
        if single_bpp.iloc[j].isnull().any():
            if single_bpp.iloc[j]["kstar_1"] == 8 and mt:
                hehg_during_mt[i] = True
                break
        
        if single_bpp.iloc[j]["evol_type"] == 3:
            mt = True
        elif single_bpp.iloc[j]["evol_type"] == 3:
            mt = False
            
print(all(hehg_during_mt))

>>> True
```

**Comment by TomWagg** [comment]
[nans_initC.csv](https://github.com/COSMIC-PopSynth/COSMIC/files/13272643/nans_initC.csv)
[nans_bpp.csv](https://github.com/COSMIC-PopSynth/COSMIC/files/13272647/nans_bpp.csv)

**Comment by TomWagg** [comment]
Here's 100 more 

[nans_bpp(1).csv](https://github.com/COSMIC-PopSynth/COSMIC/files/13283715/nans_bpp.1.csv)
[nans_initC(1).csv](https://github.com/COSMIC-PopSynth/COSMIC/files/13283716/nans_initC.1.csv)

**Comment by TomWagg** [comment]
As a test I also checked the successful binaries in the simulation (~1.5 million) and found that there were exactly zero that evolved onto HeHG (kstar_1=8) during _stable_ mass transfer. I did find successful binaries that

1. ~48,000 went through mass transfer whilst kstar_1=8, _but_ they evolved to HeHG _before_ mass transfer started
2. A further ~5000 did evolve to HeHG _during_ mass transfer, but it was during a common envelope

So in summary it seems like COSMIC can't currently handle a star evolving into HeHG _during_ **stable** mass transfer I think.
